packages/tenant-process/src/services/tenantService.ts

```diff
@@ -449,6 +456,116 @@ export function tenantServiceBuilder(
       return updatedTenant;
     },
 
+    async verifyVerifiedAttribute({
+      tenantId,
+      tenantAttributeSeed,
+      organizationId,
+      correlationId,
+    }: {
+      tenantId: TenantId;
+      tenantAttributeSeed: ApiVerifiedTenantAttributeSeed;
+      organizationId: TenantId;
+      correlationId: string;
+    }): Promise<Tenant> {
+      logger.info(
+        `Verifying attribute ${tenantAttributeSeed.id} to tenant ${tenantId}`
+      );
+
+      if (organizationId === tenantId) {
+        throw verifiedAttributeSelfVerification();
+      }
+
+      const allowedStatuses = [
+        agreementState.pending,
+        agreementState.active,
+        agreementState.suspended,
+      ];
+      await assertVerifiedAttributeOperationAllowed({
+        producerId: organizationId,
+        consumerId: tenantId,
+        attributeId: unsafeBrandId(tenantAttributeSeed.id),
+        agreementStates: allowedStatuses,
+        readModelService,
+        error: attributeVerificationNotAllowed(
+          tenantId,
+          unsafeBrandId(tenantAttributeSeed.id)
+        ),
+      });
+
+      const targetTenant = await retrieveTenant(tenantId, readModelService);
+
+      const verifiedTenantAttribute = targetTenant.data.attributes.find(
+        (attr): attr is VerifiedTenantAttribute =>
+          attr.type === tenantAttributeType.VERIFIED &&
+          attr.id === tenantAttributeSeed.id
+      );
+
+      // eslint-disable-next-line functional/no-let
+      let updatedTenant: Tenant = {
+        ...targetTenant.data,
+        updatedAt: new Date(),
+      };
+      if (!verifiedTenantAttribute) {
+        // assigning attribute for the first time
+        updatedTenant = {
+          ...updatedTenant,
+          attributes: [
+            {
+              id: unsafeBrandId(tenantAttributeSeed.id),
+              type: tenantAttributeType.VERIFIED,
+              assignmentTimestamp: new Date(),
+              verifiedBy: [
+                {
+                  id: organizationId,
+                  verificationDate: new Date(),
+                  expirationDate: tenantAttributeSeed.expirationDate
+                    ? new Date(tenantAttributeSeed.expirationDate)
+                    : undefined,
+                  extensionDate: tenantAttributeSeed.expirationDate
+                    ? new Date(tenantAttributeSeed.expirationDate)
+                    : undefined,
+                },
+              ],
+              revokedBy: [],
+            },
```

Comment:
In this way all other attributes are deleted. This is not the desired behavior: the new attribute should be appended to the array.